osm2streets/src/marking.rs

```diff
@@ -1,46 +1,117 @@
+// We use geom and stay in map space. Output is done in latlon.
+use geom::{Angle, Line, PolyLine, Polygon, Pt2D};
+
+use crate::lanes::TrafficMode;
+use crate::marking::Area::OutOfBounds;
+use crate::LaneType;
+
 pub enum Marking {
-    Longitudinal(geom::PolyLine, LongitudinalMarking),
-    Transverse(geom::Line, TransverseMarking),
-    Symbol(geom::Pt2D, geom::Angle,  SymbolMarking),
-    // Area(AreaMarking),
+    /// Lines along a lane.
+    Longitudinal(PolyLine, Longitudinal),
+    /// Lines across a lane.
+    Transverse(Line, Transverse),
+    /// Iconic or textual symbols displayed at some angle.
+    Symbol(Pt2D, Angle, Symbol),
+    /// Designated areas, like buffers, painted medians, keep clear, etc.
+    Area(Polygon, Area),
+}
+
+pub struct Longitudinal {
+    pub kind: LaneEdgeKind,
+    /// The two lanes, ltr.
+    pub lanes: [LaneType; 2],
 }
 
-pub enum LongitudinalMarking {
-    Separation {
+pub enum LaneEdgeKind {
+    OncomingSeparation {
         overtake_left: bool,
         overtake_right: bool,
     },
-    LaneDivider {
-        overtake: bool,
+    LaneSeparation {
+        merge_left: bool,
+        merge_right: bool,
     },
     RoadEdge,
-    /// Longitudinal marking that is interrupted by an intersection or merging traffic.
+    /// Longitudinal marking that is interrupted by other traffic.
     Continuity,
 }
 
-pub enum TransverseMarking {
+pub enum Transverse {
     StopLine,
     YieldLine,
 }
 
-pub enum SymbolMarking {
+pub enum Symbol {
     TrafficMode(TrafficMode),
     TurnArrow(TurnDirections),
 }
 
+// TODO move to lanes/mod.rs
 pub struct TurnDirections {
     through: bool,
     left: bool,
     right: bool,
     slight_left: bool,
     slight_right: bool,
-    u_turn: bool,
+    reverse: bool,
+}
+impl TurnDirections {
+    pub fn through() -> Self {
```

Comment:
We could consider https://crates.io/crates/enumset or similar later to make the API less boilerplatey, but starting simple is good

Reply:
I plan on parsing turn directions (and implementing a basic turn arrow rendering :P) in a follow up PR.